web/src/main/webapp/dialog.ts

```diff
@@ -0,0 +1,86 @@
+import {IHtmlElement} from "./ui"
+import {ContentsKind, ColumnDescription} from "./table"
+import {Pair} from "./util"
+
+export class DialogField {
+	html: HTMLSelectElement | HTMLInputElement;
+	type?: ContentsKind;
+}
+
+export abstract class Dialog implements IHtmlElement {
```

Comment:
Can you also please modify the method chooseBuckets() in histogram.ts, the line using window.prompt, to use a dialog instead?